NewTutorials/Tutorial/TutorialCasesViewController.swift

```diff
@@ -2,67 +2,42 @@
 
 import UIKit
 import OneMobileSDK
-import PlayerControls
 
-func setupPlayingVideos(vc: TutorialCasesViewController) {
-    func onSuccess(player: Player) {
-        let playerViewController = PlayerViewController()
-        playerViewController.contentControlsViewController = DefaultControlsViewController()
-        playerViewController.player = player
-        vc.navigationController?.pushViewController(playerViewController, animated: true)
-    }
-    
-    vc.props = .init(
-        rows: [
-            TextCell.ViewModel.init(name: "Single video", action: {
-                vc.show(player: singleVideo().dispatch(on: .main).onSuccess(call: onSuccess))
-            }),
-            TextCell.ViewModel.init(name: "Array of videos", action: {
-                vc.show(player: arrayOfVideos().dispatch(on: .main).onSuccess(call: onSuccess))
-            }),
-            TextCell.ViewModel.init(name: "Video playlist", action: {
-                vc.show(player: videoPlaylist().dispatch(on: .main).onSuccess(call: onSuccess))
-            }),
-            TextCell.ViewModel.init(name: "Muted video", action: {
-                func mute(player: inout Player) { player.mute() }
-                vc.show(player: singleVideo().map(mute).dispatch(on: .main).onSuccess(call: onSuccess))
-            }),
-            TextCell.ViewModel.init(name: "Video without autoplay", action: {
-                vc.show(player: singleVideo(isAutoplay: false).dispatch(on: .main).onSuccess(call: onSuccess))
-            })])
-}
 
 class TutorialCasesViewController: UITableViewController {
     struct Props {
-        var rows: [CellViewModel] = []
+        var rows: [TextCell.ViewModel] = []
+        var isLoading = false
     }
     
     @IBOutlet weak private var activityIndicator: UIActivityIndicatorView!
     
-    var props: Props = Props()
-    
-    func show(player: Future<Result<Player>>) {
-        view.isUserInteractionEnabled = false
-        activityIndicator.isHidden = false
-        activityIndicator.startAnimating()
-        
-        func stopWaiting() {
-            self.view.isUserInteractionEnabled = true
-            self.activityIndicator.stopAnimating()
+    var props: Props = Props() {
+        didSet {
+            guard isViewLoaded else { return }
+            tableView.reloadData()
+            view.isUserInteractionEnabled = !props.isLoading
+            activityIndicator.isHidden = !props.isLoading
+            props.isLoading ?
```

Comment:
All UI changes have to be done in `viewDidLayoutSubviews` - here you should only trigger `view.setNeedsLayout` and `tableView.reloadData()`